pkgs/by-name/un/unblob/package.nix

```
@@ -0,0 +1,129 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  gitUpdater,
+  makeWrapper,
+  e2fsprogs,
+  jefferson,
+  lz4,
+  lziprecover,
+  lzop,
+  p7zip,
+  sasquatch,
+  sasquatch-v4be,
+  simg2img,
+  ubi_reader,
+  unar,
+  zstd,
+}:
+
+let
+  # These dependencies are only added to PATH
+  runtimeDeps = [
+    e2fsprogs
+    jefferson
+    lziprecover
+    lzop
+    p7zip
+    sasquatch
+    sasquatch-v4be
+    ubi_reader
+    simg2img
+    unar
+    zstd
+    lz4
+  ];
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "unblob";
+  version = "24.12.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "onekey-sec";
+    repo = "unblob";
+    tag = version;
+    hash = "sha256-lQiY5CDz4y2cOGh4gYjRPUC7jLbbvredEBO/vPQN5Eo=";
+    forceFetchGit = true;
+    fetchLFS = true;
+  };
+
+  strictDeps = true;
+
+  build-system = with python3.pkgs; [ poetry-core ];
+
+  dependencies = with python3.pkgs; [
+    arpy
+    attrs
+    click
+    cryptography
+    dissect-cstruct
+    lark
+    lief.py
+    python3.pkgs.lz4 # shadowed by pkgs.lz4
+    plotext
+    pluggy
+    pyfatfs
+    pyperscan
+    python-magic
+    rarfile
+    rich
+    structlog
+    treelib
+    unblob-native
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  # These are runtime-only CLI dependencies, which are used through
+  # their CLI interface
+  pythonRemoveDeps = [
+    "jefferson"
+    "ubi-reader"
+  ];
+
+  pythonImportsCheck = [ "unblob" ];
+
+  makeWrapperArgs = [
+    "--prefix PATH : ${lib.makeBinPath runtimeDeps}"
+  ];
+
+  nativeCheckInputs =
+    with python3.pkgs;
+    [
+      pytestCheckHook
+      pytest-cov
+    ]
+    ++ runtimeDeps;
+
+  pytestFlagsArray = [
+    "--no-cov"
+    # `disabledTests` swallows the parameters between square brackets
+    # https://github.com/tytso/e2fsprogs/issues/152
+    "-k 'not test_all_handlers[filesystem.extfs]'"
+  ];
+
+  doInstallCheck = true;
+
+  nativeInstallCheckInputs = with python3.pkgs; [
+    versionCheckHook
+  ];
```

I'm not sure how this worked before
```suggestion
  nativeInstallCheckInputs = [
    versionCheckHook
  ];
```

---

I am not sure when and how this hook would run, but it doesn't seem to be executed.
If I replace this with this, the package still builds:

```nix
  nativeInstallCheckInputs = throw "nativeInstallCheckInputs";
```

---

Looks like, `builtPythonPackage` overwrites this: https://github.com/NixOS/nixpkgs/blob/5b37bd1fd84170665ea72db71ae33acb4312572d/pkgs/development/interpreters/python/mk-python-derivation.nix#L383

---

imo, we need to put it in `nativeCheckInputs`

---

Yeah it should be in `nativeCheckInputs` for Python packages but `nativeInstallCheckInputs` anywhere else